### ISSUE TITLE
ITMS-90809: App rejection due to UIWebView usage

### ISSUE THREAD

**Issue by abhishekthanvi8** [twilio/twilio-video-ios#198]
```
ITMS-90809: Deprecated API Usage - New apps that use UIWebView are no longer accepted. Instead, use WKWebView for improved security and reliability. Learn more (https://developer.apple.com/documentation/uikit/uiwebview).

Best regards,

The App Store Team
```

The app is getting rejected for an issue with the build I tried to with this command  `grep -r "UIWebView" .`
and found this

**Binary file ./Pods/TwilioVideo/TwilioVideo.xcframework/ios-i386_x86_64-simulator/dSYMs/TwilioVideo.framework.dSYM/Contents/Resources/DWARF/TwilioVideo matches
Binary file ./Pods/TwilioVideo/TwilioVideo.xcframework/ios-arm64_armv7/dSYMs/TwilioVideo.framework.dSYM/Contents/Resources/DWARF/TwilioVideo matches**

How to fix this?

**Comment by ceaglest** [comment]
Hi @abhishekthanvi8,

I'm sorry to hear about the app rejection, we are investigating it.

Our SDK doesn't explicitly use UIWebView, but that being said it appears that WebRTC has some internal references to it. Could you provide a little more information requested in the issue template?

1. Which version of our SDK are you using? (It looks like 4.x).
2. Does your app require iOS < 11 support (the armv7 and x86_64 slices)?

Best,
Chris

**Comment by ceaglest** [comment]
Hello,

In 4.0 we began packaging Twilio Video as an XCFramework including DWARF debugging symbols (dSYM) so that users can symbolicate stack traces which include SDK code. It appears that the references to UIWebView are included in the .dSYM files even though they are not present in the main binaries. This is true for our most recent releases 4.4.0, and 4.6.0-beta1.

For now, it's possible if you can just remove the dSYM files entirely then your submission will succeed. I am investigating a fix to either strip the offending symbols from the .dSYM files or modify our build pipeline so the so the symbols are never generated in the first place.

Regards,
Chris

**Comment by abhishekthanvi8** [comment]
How to remove dSYM files? Any steps?

**Comment by abhishekthanvi8** [comment]
1. Which version of our SDK are you using? (It looks like 4.x). **YES IT IS 4.0.0**
2. Does your app require iOS < 11 support (the armv7 and x86_64 slices)? **NO, IT REQUIRES IOS SUPPORT 12**

**Comment by ceaglest** [comment]
Hi @abhishekthanvi8,

Thank you for the info.  Based upon this and my own checks of the artifacts I can see that all versions since 4.0.0 are impacted by this issue.

> How to remove dSYM files? Any steps?

I haven't been able to confirm yet but you might just be able to delete each of the .dSYM files while we work on a fix.

```
./Pods/TwilioVideo/TwilioVideo.xcframework/ios-i386_x86_64-simulator/dSYMs/TwilioVideo.framework.dSYM
./Pods/TwilioVideo/TwilioVideo.xcframework/ios-arm64_armv7/dSYMs/TwilioVideo.framework.dSYM
```

I apologize for this unexpected app rejection. We are making it a top priority to patch the issue.

Best,
Chris

**Comment by bobiechen-twilio** [comment]
Hi @abhishekthanvi8,

Bobie here and I work with Chris on the iOS SDK team.
While @ceaglest was investigating the symbols in the dSYM file and making sure the SDK does not use any deprecated APIs, we also used our internal app (which uses the iOS SDK) and uploaded the archive for App Store submission.

During the process we tried instantiating a `UIWebView` object in the app, which did not have any use of the class previously. We then received an email from App Store Connect notifying that the app did not comply to the same **ITMS-90809: Deprecated API Usage** policy because the `UIWebView` class has been deprecated. Once we removed the `UIWebView` instance in the app and uploaded again, no more notification email was received and we were able to choose the build we just uploaded in the submission page.

This should give you the confidence that the iOS SDK does not contain any API that violates Apple's policy and is safe to use. This also makes us think that the usage of `UIWebView` might come from other libraries or frameworks used in the app. You can `otool` or `strings` to validate the symbols and locate the deprecated class in the dependencies.

Best,
Bobie

**Comment by abhishekmukherg** [comment]
I think you meant for @abhishekthanvi8  (just in case it didn't ping them)

**Comment by ceaglest** [comment]
Hi @abhishekthanvi8,

Did Bobie's response help you get to the bottom of the UIWebView usage? 

I did attempt to eliminate the references to UIWebView in the .dSYM files with narrowly scoped includes but I did not succeed. I think we have done the best we can which is to determine that our binary files do not include any references to UIWebView and that we are able to submit an app using the Voice & Video SDKs.

Best,
Chris
